cmd/internal/opentelemetry/wrapperMux.go

```diff
@@ -0,0 +1,203 @@
+	providerOpts := []trace.ProviderOption{
+		trace.WithBatcher(exporter),
+		trace.WithConfig(trace.Config{DefaultSampler: sampler}),
+	}
+
+	traceProvider, err := trace.NewProvider(providerOpts...)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	global.SetTraceProvider(traceProvider)
+
+	apiB3Single, apiB3, traceCtx, corrCtx := apiTrace.B3{SingleHeader: false}, apiTrace.B3{SingleHeader: false}, apiTrace.DefaultHTTPPropagator(), correlation.DefaultHTTPPropagator()
+	extractor := extractorFilter{[]propagation.HTTPExtractor{apiB3Single, apiB3, traceCtx, corrCtx}}
+	global.SetPropagators(propagation.New(propagation.WithInjectors(apiB3, traceCtx, corrCtx), propagation.WithExtractors(extractor)))
+	return exporter
+}
+
+type extractorFilter struct {
+	extractors []propagation.HTTPExtractor
+}
+
+func (e extractorFilter) Extract(orgCtx context.Context, supplier propagation.HTTPSupplier) context.Context {
+	for _, extractor := range e.extractors {
+		if newCtx := extractor.Extract(orgCtx, supplier); apiTrace.RemoteSpanContextFromContext(newCtx).IsValid() {
+			return newCtx
+		}
+	}
+	return orgCtx
+}
+
+func getTraceSamplerProbability() trace.Sampler {
+	probability := *traceProbability
+	if probability < 0 && probability > 1 {
+		klog.Errorf("invalid probability value %f, probability is allowed between 0.0 and 1.0", probability)
+	}
+	if probability == 0 {
+		klog.Info("set never sampler")
+		return trace.NeverSample()
+	} else if probability == 1 {
+		klog.Info("set always sampler")
+		return trace.AlwaysSample()
+	} else {
+		klog.Info("set probability sampler")
+		return trace.ProbabilitySampler(probability)
+	}
+}
+
+func WrapperServerMux(mux *http.ServeMux) *wrapperMux {
+	return &wrapperMux{mux: mux}
+}
+
+type wrapperMux struct {
+	mux *http.ServeMux
+}
+
+func (m *wrapperMux) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	m.mux.ServeHTTP(writer, request)
+}
+
+func (m *wrapperMux) HandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request)) {
+	wH := wrapperHandler{callback: handler}
+	m.mux.Handle(pattern, othttp.NewHandler(&wH, pattern))
+}
+
+func (m *wrapperMux) Handler(r *http.Request) (http.Handler, string) {
+	h, p := m.mux.Handler(r)
+	return othttp.NewHandler(h, p), p
+}
+
+func (m *wrapperMux) Handle(pattern string, handler http.Handler) {
+	m.mux.Handle(pattern, othttp.NewHandler(handler, pattern))
+}
+
+type wrapperHandler struct {
+	callback func(http.ResponseWriter, *http.Request)
+}
+
+func (w *wrapperHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	if w.callback != nil {
+		w.callback(writer, request)
+	} else {
+		panic(fmt.Errorf("not yet implemented"))
+	}
+}
+
+func getTraceTags(tags []string) []kv.KeyValue {
+	keyValue := []kv.KeyValue{
+		kv.String("service.name", traceServiceName),
+		kv.String("service.version", fmt.Sprintf("semver: %s git:%s", version.Info["version"], version.Info["revision"])),
```

Comment:
I think the idea is either use one or another, not two together. Do you need both?